scene/2d/tile_map_layer.cpp

```diff
@@ -2187,6 +2189,7 @@ void TileMapLayer::set_cell(const Vector2i &p_coords, int p_source_id, const Vec
 	c.alternative_tile = alternative_tile;
 
 	// Make the given cell dirty.
+	MutexLock lock(dirty.mutex);
 	if (!E->value.dirty_list_element.in_list()) {
 		dirty.cell_list.add(&(E->value.dirty_list_element));
 	}
```

Comment:
```suggestion
	{
		MutexLock lock(dirty.mutex);
		if (!E->value.dirty_list_element.in_list()) {
			dirty.cell_list.add(&(E->value.dirty_list_element));
		}
	}
```
I'd at least reduce this one, the rest should be looked into

Reply:
Though there *is* a possible race condition between this whole method and the update one, as it looks up an iterator in `tile_map` above which is touched by the update, I think this whole class needs some thread safety evaluation

Reply:
There *is* threading for the `TileMap` editor and specifically the pattern generation, I suspect this is what's causing this and might be the area to look into, rather than adding it to the `TileMap` itself, unsure how to handle that part specifically though

Reply:
I suspect this code is the problem:
https://github.com/godotengine/godot/blob/0bcc0e92b3f0ac57d4c4650722f347593a258572/editor/plugins/tiles/tiles_editor_plugin.cpp#L70-L144

But unsure how to approach it exactly, might work by adding a `update_internals()` after setting the pattern, or doing some other deferred operation here, will check if I can replicate the crash later today and see about this as a solution